src/Database/LSMTree/Internal/PageAcc.hs

```diff
@@ -205,6 +211,9 @@ resetPageAcc PageAcc {..} = do
     -- P.setPrimArray paBlobRefs1 0 maxKeys 0
     -- P.setPrimArray paBlobRefs1 0 maxKeys 0
 
+    VM.mapM_ (\_ -> pure $! dummyKey) paKeys
+    VM.mapM_ (\_ -> pure $! dummyValue) paValues
```

Comment:
We can do less work, by only overwriting up to it's previous size. These arrays are quite large.

And we can use `VM.set` rather than `mapM_`. So something like `VM.set dummy (VM.slice ...)`